pkg/controllers/cspi-controller/handler.go

```diff
@@ -241,6 +227,51 @@ func (c *CStorPoolInstanceController) updateStatus(cspi *cstor.CStorPoolInstance
 	return nil
 }
 
+// updateROMode sets/unsets the pool readonly mode property. It does the following changes
+// 1. If pool used space reached to roThresholdLimit then pool will be set to readonly mode
+// 2. If pool was in readonly mode if roThresholdLimit/pool expansion was happened then it
+//    unsets the ReadOnly Mode.
+// NOTE: This function must be invoked after having the updated cspiStatus information
+func (c *CStorPoolInstanceController) updateROMode(
+	cspiStatus *cstor.CStorPoolInstanceStatus, cspi cstor.CStorPoolInstance) {
+	roThresholdLimit := cspi.Spec.PoolConfig.ROThresholdLimit
+	totalInBytes := cspiStatus.Capacity.Total.Value()
+	usedInBytes := cspiStatus.Capacity.Used.Value()
+	pool := zpool.PoolName()
+
+	usedPercentage := (usedInBytes * 100) / totalInBytes
+	if (int(usedPercentage) >= roThresholdLimit) &&
+		(roThresholdLimit != 0 && roThresholdLimit != 100) {
```

Comment:
I will rather say -- 
1. `roThresholdLimit==0` should mark pool as read only. 
Me as a user -- I want to set the pool to read only (why ? ) . How can I do that ? 
Well edit your cspc and put `roThreshold` to 0.

Then, how we differentiate b/w 0 `roThreshold` and not specified  `roThreshold`
Well then -- lets make  ` roThreshold` a pointer field.

2. And `roThresholdLimit==100` by defaults fits into maths for putting pool into RO ? 
is there a situation when the pool is exhausted 100% and still we do not want to put it in RO as we do not support unmap and want to rewrite ?

Reply:
> roThresholdLimit==0 should mark pool as read only.
Me as a user -- I want to set the pool to read only (why ? ) . How can I do that ?
Well edit your cspc and put roThreshold to 0.

I also thought of treating it as a pointer but I am not aware of any such use case to honor zero value of roThresholdLimit.

Reply:
> And roThresholdLimit==100 by defaults fits into maths for putting pool into RO ?
is there a situation when the pool is exhausted 100% and still we do not want to put it in RO as we do not support unmap and want to rewrite ?

If the user wants to use the entier pool we shouldn't stop the user but anyhow, in reality, 100% pool can't be consumed even though if it consumed we can't perform any operations on the pool until expansion happens. So If user-configured 100% we shouldn't touch the pool will be automatically transferred to some other state(something like non-operational mode).